darts/timeseries.py

```diff
@@ -911,6 +935,160 @@ def univariate_values(self, copy=True, sample=0) -> np.ndarray:
+        """Returns correct axis name as a string, regardless whether it was string or integer"""
+        axis_error_string = "Axis parameter can be only one of the numbers (0, 1, 2) " \
+                            "or strings ('time', 'component', 'sample')"
+        if isinstance(axis, int):
+            try:
+                axis = DIMS[axis]
+            except IndexError:
+                raise AttributeError(axis_error_string)
+        elif isinstance(axis, str):
+            if axis not in DIMS:
+                raise AttributeError(axis_error_string)
+        else:
+            raise AttributeError(axis_error_string)
+
+        return axis
+
+    @staticmethod
+    def concatenate(timeserie_sequence: Sequence['TimeSeries'],
+                    axis: Optional[Union[str, int]] = 0,
+                    ignore_time_axes: Optional[bool] = False):
+        """Concatenates other timeseries to the current one along given axis.
+
+            Note: when concatenating along the ``time`` dimension, first concatenated timeserie marks the start date of
+            concatenated series and remaining series will have their date indices overwritten.
+
+            Parameters
+            ----------
+            timeserie_sequence : sequence of TimeSeries
+                another timeseries to concatenate to this one
+            axis : str or int
+                axis along which timeseries will be concatenated. ['time', 'component' or 'sample'; Default: 'time']
+            ignore_time_axes : bool, default False
+                Ignore errors when time axis is varies for some time series. Note that this may yield unexpected results
+
+            Returns
+            -------
+            TimeSeries
+                concatenated timeseries
+        """
+
+        axis = TimeSeries._get_str_axis(axis)
+        if len(timeserie_sequence) < 2:
+            return timeserie_sequence
+
+        start_date_equal = len(set([ts.start_time() for ts in timeserie_sequence])) == 1
+        end_date_equal = len(set([ts.end_time() for ts in timeserie_sequence])) == 1
+        freq_equal = len(set([ts.freq_str for ts in timeserie_sequence])) == 1
+        time_axis_equal = start_date_equal and end_date_equal and freq_equal
+        component_axis_equal = len(set([ts.width for ts in timeserie_sequence])) == 1
+        sample_axis_equal = len(set([ts.n_samples for ts in timeserie_sequence])) == 1
+
+        if ((axis == 'time' and not (component_axis_equal and sample_axis_equal)) or
+            (axis == 'component' and not ((time_axis_equal or ignore_time_axes) and sample_axis_equal)) or
+            (axis == 'sample' and not ((time_axis_equal or ignore_time_axes) and component_axis_equal))):
+
+            raise_log(AttributeError('Remaining (non-concatenating) axes need to be equal.'))
+
+        da_sequence = [ts.data_array() for ts in timeserie_sequence]
+
+        if axis == DIMS[0]: # time
+            if time_axis_equal:
```

Comment:
I don't really think you need to explicitly check this case here, because it will later be assessed that `consecutive_time_axes == True`, and `time_axis_equal` is just a sub case of time axes not being consecutive but I don't think we need to check for it and adapt the error message in this case.

Reply:
Not exactly: in the former case, `ignore_time_axes` can be used to override it and return only the first timeseries. In the latter case, `ignore_time_axes` doesn't do anything (and maybe it shouldn't be this way). Should I change the behavior to one case, where `ignore_time_axes` returns first series only, whether they perfectly overlap or not?